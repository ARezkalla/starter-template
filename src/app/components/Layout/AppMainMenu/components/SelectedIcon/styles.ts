export const styles = {
  icon: {
    color: "text.primary",
  },
  iconSelected: {
    color: "primary.light",
  },
}
