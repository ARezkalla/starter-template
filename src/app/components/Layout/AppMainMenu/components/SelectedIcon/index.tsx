import React from "react"

import { styles } from "./styles"
import { AppIcon } from "screens/Facilities/subcomponents/Dynamic/SimpleTimeseriesGraph/subcomponents/GraphBaseHeader/subcomponents/AppIcon"

interface SelectedIconProps {
  iconCode: any
  isActive: boolean
}

const SelectedIcon = ({ iconCode, isActive }: SelectedIconProps) => (
  <AppIcon
    iconCode={iconCode}
    sx={{
      ...styles.icon,
      ...(isActive && styles.iconSelected),
    }}
  />
)

export default SelectedIcon
