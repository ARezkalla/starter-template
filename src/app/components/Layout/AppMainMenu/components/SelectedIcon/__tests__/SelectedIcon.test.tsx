import { renderWithProviders } from "@/app/test/renderWithProvider"
import SelectedIcon from "../"
import { it, describe } from "vitest"

describe("SelectedIcon", () => {
  it("should render", () => {
    const { container } = renderWithProviders(
      <SelectedIcon iconCode="code" isActive={true} />,
    )
    expect(container).toMatchSnapshot()
  })
})
