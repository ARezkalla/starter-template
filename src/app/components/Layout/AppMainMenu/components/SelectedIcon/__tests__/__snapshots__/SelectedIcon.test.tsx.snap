// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SelectedIcon > should render 1`] = `
<div>
  <div
    class="MuiBox-root css-5azvwx"
  >
    <span
      aria-hidden="true"
      class="material-icons notranslate MuiIcon-root MuiIcon-fontSizeMedium css-1ka11ga-MuiIcon-root"
    >
      code
    </span>
  </div>
</div>
`;
